yb-voyager/cmd/importDataSequenceRestoration.go

```diff
@@ -0,0 +1,291 @@
+		}
+		sequenceAttachedToTables, ok := sequenceNameToTableMap.Get(sequenceTuple)
+		if !ok {
+			//sequence is not attached to any table, right now this is not expected from export as we only migrate sequences
+			// attached to tables in all migration workflows
+			log.Infof("sequence %q is not attached to any table, skipping", sequenceName)
+			continue
+		}
+		//check if the sequence is attached to a table and present in the table list
+		isTablePresentInTableList, err := checkIfSequenceAttachedToTablesInTableList(importTableList, sequenceAttachedToTables)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if sequence is attached to tables in table list: %w", err)
+		}
+		if !isTablePresentInTableList {
+			//if the sequence is attached to a table but not in the table list
+			//skip the sequence
+			log.Infof("sequence %q is not present in the table list, skipping", sequenceName)
+			continue
+		}
+		if !sequenceTuple.TargetTableAvailable() {
+			//if the sequence is attached to a table and table in table list but sequence not present in the target database
+			//return error
+			return nil, fmt.Errorf("sequence %q is not present in the target database", sequenceName)
+		}
+		//restore the sequence last value
+		sequenceNameTupleToLastValueMap.Put(sequenceTuple, lastValue)
+
+	}
+	return sequenceNameTupleToLastValueMap, nil
+}
+
+func readSequenceLastValueFromPostDataSql(sequenceFilePath string, sourceDBType string) (map[string]int64, error) {
+	sequenceLastValue := make(map[string]int64)
+	//get all the setval statements from the postdata.sql file
+	sqlInfoArr := parseSqlFileForObjectType(sequenceFilePath, "SEQUENCE")
+	for _, sqlInfo := range sqlInfoArr {
+		parseTree, err := queryparser.Parse(sqlInfo.stmt)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing the ddl[%s]: %v", sqlInfo.stmt, err)
+		}
+		var sequenceName string
+		var lastValue int64
+		switch sourceDBType {
+		case POSTGRESQL:
+			//check if the statement is a setval statement for POSTGRESQL SELECT setval('sequence_name', last_value)
+			if !queryparser.IsSelectSetValStmt(parseTree) {
+				log.Infof("not a setval statement: %s", sqlInfo.stmt)
+				continue
+			}
+			//get the sequence name and last value from the setval statement
+			sequenceName, lastValue, err = queryparser.GetSequenceNameAndLastValueFromSetValStmt(parseTree)
+			if err != nil {
+				return nil, fmt.Errorf("error getting sequence name and last value from setval statement: %w", err)
+			}
+		case ORACLE, MYSQL:
+			//in case of ORACLE and MYSQL, we need to check if its as ALTER SEQUENCE seq RESTART 11 because ora2pg dumps it like that
+			if !queryparser.IsAlterSequenceStmt(parseTree) {
+				log.Infof("not an alter sequence statement: %s", sqlInfo.stmt)
+				continue
+			}
+			//get the sequence name and last value from the alter sequence statement
+			sequenceName, lastValue, err = queryparser.GetSequenceNameAndLastValueFromAlterSequenceStmt(parseTree)
```

Comment:
Just to be clear, let's change the name of func `GetSequenceNameAndRestartValueFromAlterSequenceStmt`
and add a comment about why it might be okay to assume restart value as the last value.